### PR TITLE
[FIX] web: fix smooth scroll while dragging into HTML element

### PR DESCRIPTION
**Pull request by bvr-odoo**
Steps to reproduce the bug:

- In website edit mode.
- Add enough blocks to the page so that a vertical scrollbar appears.
- Drag a block and move it to the bottom of the page to scroll down.
- Bug: The page does not scroll.

This bug was introduced because the scroll element is now the `<html>`
instead of the "#wrapwrap" element (Since this commit [1]). The issue
arises because "getBoundingClientRect()" gives the distance from the top
of the element to the viewport, excluding the scroll position, while the
`<html>` element accounts for scrollTop to define its Y position.

In this commit, we also fixed the "dragStart" function, which didn't
return the HTML element as the scroll element when it was the drag and
drop container. The issue is that the "closestScrollableY" function
don't consider an element with "overflow: visible" as scrollable.

At first, we tried to modify "closestScrollableY" to consider an element
with "overflow: visible" as scrollable, but this caused several errors
in the "drag and drop" QUnit tests and we couldn't find the cause. This
needs further investigation.

To fix the issue, we updated the "dragStart" function to check if the
"drag and drop" container is the "ownerDocument.scrollingElement". If it
is, we use that element as the scrollable element for the "drag and drop".

[1]: https://github.com/odoo/odoo/commit/189a7c96e6e26825dc05c0c6466576fe63aa091e

task-4190506

### CODE REVIEW

File: addons/web/static/src/core/utils/draggable_hook_builder.js [hunk 1]
```diff
@@ -580,6 +586,15 @@ export function makeDraggableHook(hookParams) {
                 const xRect = ctx.current.scrollParentXRect;
                 const yRect = ctx.current.scrollParentYRect;
 
+                // getBoundingClientRect() (used in getRect()) gives the
+                // distance from the element's top to the viewport, excluding
+                // scroll position. Only the <html> element accounts for
+                // scrollTop.
+                const scrollParentYEl = ctx.current.scrollParentY;
+                if (scrollParentYEl?.nodeName === "HTML") {
```
Comment (qsm-odoo) [inline addons/web/static/src/core/utils/draggable_hook_builder.js:594]: - Why not comparing to `ownerDocument.scrollingElement` here?
- How is it possible to have `scrollParentYEl` falsy?

File: addons/web/static/src/core/utils/draggable_hook_builder.js [hunk 2]
```diff
@@ -500,9 +500,15 @@ export function makeDraggableHook(hookParams) {
...
-                );
+                const isDocumentScrollingElement = ctx.current.container
+                    === ctx.current.container.ownerDocument.scrollingElement;
+                // If the container is the "ownerDocument.scrollingElement",
+                // there is no need to get the scroll parent as it is the
+                // scrollable element itself.
+                [ctx.current.scrollParentX, ctx.current.scrollParentY] =
+                    isDocumentScrollingElement
+                    ? [ctx.current.container, ctx.current.container]
+                    : getScrollParents(ctx.current.container);
```
Comment (qsm-odoo) [inline addons/web/static/src/core/utils/draggable_hook_builder.js:511]: Add a "// TODO investigate if `getScrollParents` should not consider the `document.scrollingElement` directly"

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/web/static/src/core/utils/draggable_hook_builder.js (hunk 1 above)
Comment (bvr-odoo) [reply addons/web/static/src/core/utils/draggable_hook_builder.js:594]: > * Why not comparing to `ownerDocument.scrollingElement` here?

Was not sure if `ownerDocument.scrollingElement` is always an HTML element

> * How is it possible to have `scrollParentYEl` falsy?

This is the case when `closestScrollableY` used in `getScrollParents` returns `NULL` (no vertical scrollbar on the page)

**Review by qsm-odoo — APPROVED** [review]
Pushed this diff: https://github.com/odoo/odoo/compare/d74690fafa29f7777485eba8d938bde32d5d39ec..d9e413f1837f31bb4278cdd644940b83123628a6

@robodoo r+

**Pull request closed without merging**